metals/src/main/scala/scala/meta/internal/metals/SemanticTokensProvider.scala

```diff
@@ -20,16 +20,48 @@ import org.eclipse.lsp4j.SemanticTokenTypes
  */
 object SemanticTokensProvider {
 
-  def getTokens(isScala3: Boolean, params: VirtualFileParams): Tokens = {
+  def getTokens(isScala3: Boolean, text: String): Tokens = {
     import scala.meta._
     if (isScala3) {
       implicit val dialect = scala.meta.dialects.Scala3
-      params.text().tokenize.get
+      text.tokenize.get
     } else {
-      params.text().tokenize.get
+      text.tokenize.get
     }
   }
 
+  private def addToBuffer(
```

Comment:
```suggestion
  private def convertTokensToIntArray(
```
?
We don't add anything to buffer here.